tests/test_single_topology_combined.py

```diff
@@ -179,3 +179,42 @@ def _test_combined_parameters_impl_nonbonded(host_system_omm):
                 assert ref_q == test_q
                 assert test_sig == ref_sig
                 assert test_eps == ref_eps
+
+
+@pytest.mark.parametrize("host_system_fixture", ["solvent_host_system", "complex_host_system"])
+def test_combined_parameters_nonbonded_intermediate(
+    host_system_fixture, hif2a_ligand_pair_single_topology: SingleTopology, request
+):
+    st = hif2a_ligand_pair_single_topology
+    host_sys, host_masses = request.getfixturevalue(host_system_fixture)
+    num_host_atoms = len(host_masses)
+
+    rng = np.random.default_rng(2022)
+
+    for lamb in rng.uniform(0.01, 0.99, (10,)):
+        hgs = st.combine_with_host(host_sys, lamb=lamb)
+        cutoff = hgs.nonbonded_host_guest.get_cutoff()
+
+        assert hgs.nonbonded_host_guest.params is not None
+        guest_params = hgs.nonbonded_host_guest.params[num_host_atoms:]
+        ws_core = [w for flag, (_, _, _, w) in zip(st.c_flags, guest_params) if flag == 0]
+        ws_a = [w for flag, (_, _, _, w) in zip(st.c_flags, guest_params) if flag == 1]
+        ws_b = [w for flag, (_, _, _, w) in zip(st.c_flags, guest_params) if flag == 2]
+
+        # core atoms fixed at w = 0
+        assert all(w == 0.0 for w in ws_core)
```

Comment:
might be good to check that the charges, sigma, epsilon are correctly set as well (and not just the w-coords), though not necessarily for this PR.